test/functional/mempool_expiry.py

```diff
@@ -27,54 +27,52 @@ class MempoolExpiryTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def test_transaction_expiry(self, timeout):
         """Tests that a transaction expires after the expiry timeout and its
         children are removed as well."""
         node = self.nodes[0]
+        wallet = MiniWallet(node)
+
+        # Add enough mature utxos to the wallet so that all txs spend confirmed coins.
+        wallet.generate(2)
+        node.generate(100)
 
         # Send a parent transaction that will expire.
-        parent_address = node.getnewaddress()
-        parent_txid = node.sendtoaddress(parent_address, 1.0)
+        parent_txid = wallet.send_self_transfer(from_node=node)['txid']
 
         # Set the mocktime to the arrival time of the parent transaction.
         entry_time = node.getmempoolentry(parent_txid)['time']
         node.setmocktime(entry_time)
 
-        # Create child transaction spending the parent transaction
-        vout = find_vout_for_address(node, parent_txid, parent_address)
-        inputs = [{'txid': parent_txid, 'vout': vout}]
-        outputs = {node.getnewaddress(): 0.99}
-        child_raw = node.createrawtransaction(inputs, outputs)
-        child_signed = node.signrawtransactionwithwallet(child_raw)['hex']
-
-        # Let half of the timeout elapse and broadcast the child transaction.
-        half_expiry_time = entry_time + int(60 * 60 * timeout/2)
+        # Let half of the timeout elapse.
+        half_expiry_time = entry_time + 3600 * timeout // 2
         node.setmocktime(half_expiry_time)
-        child_txid = node.sendrawtransaction(child_signed)
+
+        # Broadcast the child transaction spending the parent transaction.
+        parent_utxo = wallet.get_utxo()
```

Comment:
Pretty sure this is not the parent change. miniwallet should have two outputs at this point. One leftover coinbase and one change. Coin selection favors the largest output, so the coinbase one.

I might be wrong, but I think it could make sense to check that the child tx really has parent_txid as ancestor via the mempool rpc

Reply:
I added an assertion verifying this to make it clear https://github.com/bitcoin/bitcoin/pull/20276/commits/de4173bacd3a2ce6294515072cb725d5c1074cd8.

Originally I had:
```python
parent_utxo = next(filter(lambda utxo: utxo['txid'] == parent_txid, wallet._utxos))
```

but then I realized `get_utxo()` would do this it me: ```"""Return the last utxo. Can be used to get the change output immediately after a send_self_transfer"""```

Reply:
Oh, sorry my bad in that case. Though, the mempool check would still make sense.